janusgraph-backend-testutils/src/main/java/org/janusgraph/graphdb/JanusGraphTest.java

```diff
@@ -5792,18 +5792,6 @@ private static void assertElementOrderForWithoutIndex(StandardJanusGraph graph)
 
     //................................................
 
-
-    @Test
-    public void testHasNot() {
-        JanusGraphVertex v1, v2;
-        v1 = graph.addVertex();
-
-        v2 = graph.query().hasNot("abcd").vertices().iterator().next();
-        assertEquals(v1, v2);
-        v2 = graph.query().hasNot("abcd", true).vertices().iterator().next();
-        assertEquals(v1, v2);
-    }
```

Comment:
Why this test was removed from `JanusGraphTest.java`? Tests in `JanusGraphTest.java` are executed against different storages backends. I guess, it would be better to just fix this test instead of removing it

Reply:
It is moved to `JanusGraphIndexTest` because I want to enforce the existence of indexes does not alter the behavior. Example: https://github.com/JanusGraph/janusgraph/issues/1868#issuecomment-694933702

Reply:
What about behaviour without any index? As I understand, this logic was tested with use cases when there is no necessary indices

Reply:
That is also tested. See [testHasNot](https://github.com/JanusGraph/janusgraph/pull/2265/files#diff-e22cf3f57c2f67c294951e7c0318ad14f06ae190ec0237e8076784198ddb8685R251) in JanusGraphIndexTest.

Reply:
I don't have strong opinion here. `testHasNot` which you removed was tested against different storage backends (CQL, BerkleyDB, HBase, InMemory). In the new `testHasNot` method we test it with several combinations CQL or BerkleyDB as a storage backend and Lucene / Solr / ElasticSearch as index backend. Thus, now the test is executed with less storage backends. 
As I said, I don't have strong opinion here. If you think it's better to keep all that test logic in one place (`JanusGraphIndexTest`) and don't test against InMemory and HBase I'm fine with it. I thought, maybe it's better to split that test logic between `JanusGraphTest` and `JanusGraphIndexTest`. For example, test cases `property not registered in schema`, `property registered in schema` and `property registered in schema and has composite index` can easily be moved to `JanusGraphTest` where they will be executed against a bigger number of Storage backends whereas `property registered in schema and has mixed index` can be in `JanusGraphIndexTest` where it will be executed against different mixed index backends.

Reply:
That's fair. The initial thought was to avoid having similar tests in both classes, but you made a valid point. I'll move some back to JanusGraphTest.